capstone-rs/src/instruction.rs

```diff
@@ -36,6 +37,58 @@ pub type RegIdInt = u16;
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct RegId(pub RegIdInt);
 
+/// Represents how the register is accessed.
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum RegAccessType {
+    /// Operand read from memory or register.
+    ReadOnly,
+    /// Operand write from memory or register.
+    WriteOnly,
+    /// Operand read and write from memory or register.
+    ReadWrite,
+}
+
+impl RegAccessType {
+    /// Returns whether the instruction reads from the operand.
+    ///
+    /// Note that an instruction may read and write to the register
+    /// simultaneously. In this case, the operand is also considered as
+    /// readable.
+    pub fn is_readable(self) -> bool {
+        self == RegAccessType::ReadOnly || self == RegAccessType::ReadWrite
+    }
+
+    /// Returns whether the instruction writes from the operand.
+    ///
+    /// Note that an instruction may read and write to the register
+    /// simultaneously. In this case, the operand is also considered as
+    /// writable.
+    pub fn is_writable(self) -> bool {
+        self == RegAccessType::WriteOnly || self == RegAccessType::ReadWrite
+    }
+}
+
+impl TryFrom<cs_ac_type> for RegAccessType {
+    type Error = ();
+
+    fn try_from(access: cs_ac_type) -> Result<Self, Self::Error> {
+        // Check for flags other than CS_AC_READ or CS_AC_WRITE.
+        let unknown_flag_mask = !(CS_AC_READ | CS_AC_WRITE).0;
```

Comment:
Get rid of this unknown_flag_mask check; the match statement default case handles it

Reply:
Without the check, `RegAccess::try_from(1337)` returns `Some(_)`, which fails the unit test you requested to add.

Reply:
You're right; my mistake.